src/tests.rs

```diff
@@ -0,0 +1,309 @@
+    const THRESHOLD: usize = 5;
+    pub const IP_LOCAL_HOST: IpAddr = IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1));
+
+    fn setup_members_and_init_dkg<R: RngCore>(
+        num: usize,
+        non_responsives: BTreeSet<usize>,
+        rng: &mut R,
+    ) -> (Vec<NodeID>, Vec<Member>) {
+        let (ids, mut members, _addrs, map) = create_members(num, rng);
+
+        connect_and_initialize_dkg(&mut members, map, non_responsives, THRESHOLD);
+
+        (ids, members)
+    }
+
+    fn create_members<R: RngCore>(
+        num: usize,
+        mut rng: &mut R,
+    ) -> (
+        Vec<NodeID>,
+        Vec<Member>,
+        Vec<SocketAddr>,
+        HashMap<NodeID, SocketAddr>,
+    ) {
+        let mut member_vec = vec![];
+        let mut id_vec = vec![];
+        let mut socket_addr = vec![];
+        let mut map = HashMap::new();
+
+        for _ in 0..num {
+            let mut config = Config::default();
+            config.ip = Some(IP_LOCAL_HOST);
+            config.port = Some(0);
+
+            let mut member = Member::new(config, &mut rng).unwrap();
+            let id = member.id();
+            let addr = member.our_socket_addr().unwrap();
+            member_vec.push(member);
+            id_vec.push(id.clone());
+            socket_addr.push(addr);
+            map.insert(id, addr);
+        }
+
+        (id_vec, member_vec, socket_addr, map)
+    }
+
+    fn connect_and_initialize_dkg(
+        members: &mut Vec<Member>,
+        map: HashMap<NodeID, SocketAddr>,
+        non_responsives: BTreeSet<usize>,
+        threshold: usize,
+    ) {
+        for member in members.iter_mut() {
+            member.connect_to_group(map.clone())
+        }
+
+        // Wait for Nodes to connect to each other
+        sleep(Duration::from_secs(10));
+
+        let mut proposals = vec![];
+        // Initialize DKG for all the Nodes
+        for member in members.iter_mut() {
+            let proposal = member.init_dkg(threshold).unwrap();
+            proposals.push(proposal.clone());
+        }
+
+        // Non-responsive members close connections
+        for (i, member) in members.iter_mut().enumerate() {
+            if non_responsives.contains(&i) {
+                member.set_as_non_responsive();
+            }
+        }
+
+        // Wait for Non-reponsive Node to disconnect successfully
+        if !non_responsives.is_empty() {
+            sleep(Duration::from_secs(10));
+        }
+
+        // Broadcast all the messages
+        for x in 0..members.len() {
+            if !non_responsives.contains(&x) {
+                members[x].broadcast(proposals[x].clone()).unwrap();
+            }
+        }
+
+        // Wait for the Quic threads to finish the DKG process - duration increases as the number of nodes increases
+        let len = members.len();
+        if len <= 6 {
+            sleep(Duration::from_secs(60));
+        } else if len >= 7 && len <= 10 {
+            sleep(Duration::from_secs(120));
```

Comment:
maybe a comment explaining what will be the recommended waiting time for different number of members will be enough.